application/src/main/java/org/thingsboard/server/service/transport/DefaultTransportApiService.java

```diff
@@ -345,6 +347,82 @@ private ListenableFuture<TransportApiResponseMsg> handle(GetOrCreateDeviceFromGa
         }, dbCallbackExecutorService);
     }
 
+    private ListenableFuture<TransportApiResponseMsg> handle(TransportProtos.GetOrCreateDeviceFromSparkplugRequestMsg requestMsg) {
```

Comment:
This part looks like a lot of duplication of functionality from handleGetOrCreateDeviceFromGatewayRequestMsg, probably it will be better to extract a method for the same functionality.

Reply:
I agree.

Reply:
Ok